include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -699,17 +703,17 @@ __parallel_transform_scan_single_group(oneapi::dpl::__internal::__device_backend
                         ::std::integral_constant<::std::uint16_t, __num_elems_per_item>, _BinaryOperation,
                         /* _IsFullGroup= */ std::true_type, _Inclusive, _CustomName>>>()(
                     ::std::forward<_ExecutionPolicy>(__exec), std::forward<_InRng>(__in_rng),
-                    std::forward<_OutRng>(__out_rng), __n, __init, __binary_op, __unary_op);
+                    std::forward<_OutRng>(__out_rng), __n, __init, __binary_op, __unary_op, __result_and_scratch);
```

Comment:
Do we need to pass this scratch space in only to return it?  
Perhaps it is more clear if we instead return an event from the submitter, and construct the future here with the dummy variable.  It would make it more clear that it goes unused, and minimize the changes required.

Reply:
Good point. I changed the submitters to return an event and constructed the future in the outer context instead.